pkg/agent/plugin/workloadattestor/k8s/k8s_posix.go

```diff
@@ -598,6 +618,11 @@ func getPodUIDAndContainerIDFromCGroupPath(cgroupPath string) (types.UID, string
 	matches := cgroupRE.FindStringSubmatch(cgroupPath)
 	if matches != nil {
```

Comment:
I found than there are cases where CRIO-O is providing pods on cgroups.
With minikube using cri-o as runtime, 
```
Preparing Kubernetes v1.23.3 on CRI-O 1.22.1 ...
```
I got pod on cgroup 
```
9:devices:/kubepods.slice/kubepods-besteffort.slice/kubepods-besteffort-pod561fd272_d131_47ef_a01b_46a997a778f3.slice/crio-030ded69d4c98fcf69c988f75a5eb3a1b4357e1432bd5510c936a40d7e9a1198.scope
```
And attestation works with that, 

So instead of an if/else, what do you think about getting a list of regex, and validate cgroups against both regex? 
and in case we get results from both make it fails?

in case of cgroupNoPodUidRE, maybe you can it always return 2 values, where the pod(first result) is always empty

Reply:
I implemented the list of regex. In addition I moved to named groups in the regex to be more flexible with the sequence of podUID and containerid

It's interesting to see, that the behavior in our system is caused by kubeedge and not by cri-o. Sadly I could not find an easy reason for this in kubeedge`s settings or source code.